tests/components/accuweather/test_config_flow.py

```diff
@@ -0,0 +1,158 @@
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        side_effect=RequestsExceededError(
+            "The allowed number of requests has been exceeded"
+        ),
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["errors"] == {CONF_API_KEY: "requests_exceeded"}
+
+
+async def test_integration_already_exists(hass):
+    """Test we only allow a single config flow."""
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        return_value=json.loads(load_fixture("accuweather/location_data.json")),
+    ):
+        MockConfigEntry(
+            domain=DOMAIN, unique_id="123456", data=VALID_CONFIG,
+        ).add_to_hass(hass)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "single_instance_allowed"
+
+
+async def test_create_entry(hass):
+    """Test that the user step works."""
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        return_value=json.loads(load_fixture("accuweather/location_data.json")),
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "abcd"
+        assert result["data"][CONF_NAME] == "abcd"
+        assert result["data"][CONF_LATITUDE] == 55.55
+        assert result["data"][CONF_LONGITUDE] == 122.12
+        assert result["data"][CONF_API_KEY] == "32-character-string-1234567890qw"
+
+
+async def test_options_flow(hass):
+    """Test config flow options."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, unique_id="123456", data=VALID_CONFIG,
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        return_value=json.loads(load_fixture("accuweather/location_data.json")),
+    ), patch(
+        "accuweather.AccuWeather.async_get_current_conditions",
+        return_value=json.loads(
+            load_fixture("accuweather/current_conditions_data.json")
+        ),
+    ):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        result = await hass.config_entries.options.async_init(config_entry.entry_id)
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+
+        result = await hass.config_entries.options.async_configure(
+            result["flow_id"], user_input={CONF_FORECAST: True}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert config_entry.options == {CONF_FORECAST: True}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)